sunkit_pyvista/plotter.py

```diff
@@ -174,6 +175,36 @@ def plot_solar_axis(self, length=2.5, arrow_kwargs={}, **kwargs):
                          **defaults)
         self.plotter.add_mesh(arrow, **kwargs)
 
+    def plot_quadrangle(self, m, bottom_left, top_right=None, width: u.deg = None, height: u.deg = None, **kwargs):
+        """
+        Plots a quadrangle on the given map.
+        This draws a quadrangle that has corners at ``(bottom_left, top_right)``,
+        if ``width`` and ``height`` are specified, they are respectively added to the
+        longitude and latitude of the ``bottom_left`` coordinate to calculate a
+        ``top_right`` coordinate.
+
+        Parameters
+        ----------
+        bottom_left :`~astropy.coordinates.SkyCoord`
+            The bottom-left coordinate of the quadrangle. It can
+            have shape ``(2,)`` to simultaneously define ``top_right``.
+        top_right : `~astropy.coordinates.SkyCoord`
+            The top-right coordinate of the quadrangle.
+        width : `astropy.units.Quantity`, optional
+            The width of the quadrangle. Required if ``top_right`` is omitted.
+        height : `astropy.units.Quantity`
+            The height of the quadrangle. Required if ``top_right`` is omitted.
+        **kwargs : Keyword arguments are handed to `pyvista.Plotter.add_mesh`.
+        """
+        quadrangle_patch = m.draw_quadrangle(bottom_left=bottom_left, top_right=top_right,
```

Comment:
This is not ideal, as we don't want to have a map as an input argument. I propose that instead we just copy the logic from `draw_quadrangle` directly into this function to get the patch:
```python
bottom_left, top_right = get_rectangle_coordinates(
            bottom_left, top_right=top_right, width=width, height=height)

        width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
        height = top_right.spherical.lat - bottom_left.spherical.lat

        kwergs = {
            "transform": axes.get_transform(bottom_left.frame.replicate_without_data()),
            "edgecolor": "white",
            "fill": False,
        }
        kwergs.update(kwargs)
        quad = Quadrangle(self._get_lon_lat(bottom_left), width, height, **kwergs)
```

Reply:
We would need to have `get_rectangle_coordinages()`and `Quadrangle`  imported then right?
Also, `self` is a reference to a map object, how do we handle that?

Reply:
Yep, we would need those imports. For _get_lon_lat, I think you can replace it with
```python
bottom_left = bottom_left.spherical
quad = Quadrangle((bottom_left.lon, bottom_left.lat), width, height, **kwergs)
```

Reply:
Sure, thanks!